src/runner.ts

```diff
@@ -31,9 +48,121 @@ export class ActionRunner {
     return configFile;
   }
 
+  /**
+   * The action evaluates if the rules requirements are meet for a PR
+   * @returns a true/false statement if the rule failed. This WILL BE CHANGED for an object with information (see issue #26)
+   */
+  async validatePullRequest({ rules }: ConfigurationFile): Promise<boolean> {
+    for (const rule of rules) {
+      // We get all the files that were modified and match the rules condition
+      const files = await this.listFilesThatMatchRuleCondition(rule);
+      // We check if there are any matches
+      if (files.length === 0) {
+        this.logger.debug(`Skipping rule ${rule.name} as no condition matched`);
+        // If there are no matches, we simply skip the check
+        continue;
+      }
+      if (rule.type === "basic") {
+        const [result, missingData] = await this.evaluateCondition(rule);
+        if (!result) {
+          this.logger.error(`Missing the reviews from ${JSON.stringify(missingData.missingUsers)}`);
+          return false;
+        }
+      }
+    }
+
+    // TODO: Convert this into a list of users/teams missing and convert the output into a nice summary object -> Issue #26
+    return true;
+  }
+
+  /** Evaluates if the required reviews for a condition have been meet
+   * @param rule Every rule check has this values which consist on the min required approvals and the reviewers.
+   * @returns a [bool, error data] tuple which evaluates if the condition (not the rule itself) has fulfilled the requirements
+   * @see-also ReviewError
+   */
+  async evaluateCondition(rule: { min_approvals: number } & Reviewers): Promise<ReviewError> {
```

Comment:
ReviewError sounds to me like ReviewError extends Error {...} class )) 
then I see `return [true];` 
Perhaps it could be named `ReviewState` or similar relevant to it?